grpcurl.go

```diff
@@ -645,21 +645,6 @@ func BlockingDial(ctx context.Context, network, address string, creds credential
 		writeResult:          writeResult,
 	}
 
-	dialer := func(ctx context.Context, address string) (net.Conn, error) {
-		// NB: We *could* handle the TLS handshake ourselves, in the custom
-		// dialer (instead of customizing both the dialer and the credentials).
-		// But that requires using insecure.NewCredentials() dial transport
-		// option (so that the gRPC library doesn't *also* try to do a
-		// handshake). And that would mean that the library would send the
-		// wrong ":scheme" metaheader to servers: it would send "http" instead
-		// of "https" because it is unaware that TLS is actually in use.
-		conn, err := (&net.Dialer{}).DialContext(ctx, network, address)
-		if err != nil {
-			writeResult(err)
```

Comment:
@jhump you probably have the most context on the custom dialer... maybe there was a desire to intercept dial errors here?

Reply:
Thanks for the quick feedback @dragonsinth! I had a similar thought, but I believe any dialer errors would be handled in a similar fashion (via `writeResult()`) [here](https://github.com/fullstorydev/grpcurl/blob/master/grpcurl.go#L675-L682).

Unfortunatley the default `grpc-go` [dialer](https://github.com/grpc/grpc-go/blob/master/internal/transport/proxy.go) is in an internal package, so if there is a desire to keep this error handling here I think the `grpc-go` dialer code would need to be ported over to `grpcurl`. I actually started down that path before opting for this approach. I would be happy to submit that PR as well if it would help evaluate how to address this feature request.

Reply:
Let's see if jhump has some "context" on this :)

Reply:
Hi @dragonsinth and @jhump, I went ahead and posted PR #481 which hopefully helps determine a path forward for proxy support `gprcurl`. Thanks!